routes/socket/routes.js

```diff
@@ -142,6 +200,10 @@ module.exports = (modUserNames, editorUserNames, adminUserNames) => {
 				handleUpdatedTruncateGame(data);
 			})
 			.on('addNewGameChat', data => {
+				if (isRestricted) {
+					checkRestriction();
+					if (isRestricted) return;

```

Comment:
While uh I'll allow it this sort of side-effecty code is not really the way to do things nowadays, I'm not sure this is going to work 100% the way you'd expect.  I'll take a close look tomorrow probably.

Reply:
I'm aware that isRestricted will almost always still be false and didn't bother changing it, but the intent is to update their restricted status. Most of the time, it should be lifted before they ever hit that point, unless they are doing something to bypass the notices.

Reply:
On further checking, it's a useless call. I've removed it.